src/d3a/models/myco_matcher/external_matcher.py

```diff
@@ -0,0 +1,137 @@
+import json
+
+import d3a.constants
+from d3a.d3a_core.redis_connections.redis_area_market_communicator import ResettableCommunicator
+from d3a.models.market import validate_authentic_bid_offer_pair
+from d3a.models.market.market_structures import BidOfferMatch, offer_from_JSON_string, \
+    bid_from_JSON_string
+from d3a.models.myco_matcher.base_matcher import BaseMatcher
+
+
+class ExternalMatcher(BaseMatcher):
+    def __init__(self):
+        super(ExternalMatcher, self).__init__()
+        self.simulation_id = d3a.constants.COLLABORATION_ID
+        self.myco_ext_conn = None
+        self.channel_prefix = f"external-myco/{self.simulation_id}/"
+        self._setup_redis_connection()
+        self.area_uuid_markets_mapping = {}
+        self.markets_mapping = {}
+        self.recommendations = []
+
+    def _setup_redis_connection(self):
+        self.myco_ext_conn = ResettableCommunicator()
+        self.myco_ext_conn.sub_to_multiple_channels(
+            {"external-myco/get_simulation_id": self.get_simulation_id,
+             f"{self.channel_prefix}get_offers_bids/": self.publish_offers_bids,
+             f"{self.channel_prefix}post_recommendations/": self.match_recommendations})
+
+    def publish_offers_bids(self, message):
+        """
+        Function that queries publishes open offers and bids
```

Comment:
Nit, if you're intersted: if you want to follow the PEP recommendations (from [PEP 257](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings)):
> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".

So this could become:
```suggestion
        """Query published open offers and bids.
        
        Published data are of the following format: ...
```
(The blank line in between is also part of the PEP for multiline docstrings).